### PR TITLE
Refactor and update unit-tests

### PR DESCRIPTION
**Pull request by IvanZosimov**
**Description:**
In the scope of this PR the unit-tests were refactored and updated in order to remove their integration like behavior. Dotnet installation scripts were updated as well.

**Related issue:**
https://github.com/actions/runner-images-internal/issues/4973

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: __tests__/installer.test.ts [hunk 1]
```diff
@@ -1,298 +1,402 @@
...
+    const whichSpy = jest.spyOn(io, 'which');
+    const maxSatisfyingSpy = jest.spyOn(semver, 'maxSatisfying');
+
+    describe('installDotnet() tests', () => {
+      whichSpy.mockImplementation(() => Promise.resolve('PathToShell'));
+
+      it('should throw the error in case of non-zero exit code of the installation script. The error message should contain logs.', async () => {
+        const inputVersion = '3.1.100';
+        const inputQuality = '' as QualityOptions;
+        const errorMessage = 'Fictious error message!';
```
Comment (MaksimZhukov) [inline __tests__/installer.test.ts:35]:
```suggestion
        const errorMessage = 'Fictitious error message!';
```

File: __tests__/installer.test.ts [hunk 2]
```diff
@@ -1,298 +1,402 @@
...
+          const scriptArguments = (
+            getExecOutputSpy.mock.calls[0][1] as string[]
+          ).join(' ');
+
+          expect(scriptArguments).toContain(
+            `-ProxyBypassList ${process.env['no_proxy']}`
+          );
+        });
+      }
     });
```
Comment (MaksimZhukov) [inline __tests__/installer.test.ts:254]:
```suggestion
    });

```

**Review by MaksimZhukov — APPROVED** [review]

File: __tests__/installer.test.ts (hunk 1 above)
Comment (IvanZosimov) [reply __tests__/installer.test.ts]: Updated this typo everywhere in the tests.

File: __tests__/installer.test.ts (hunk 2 above)
Comment (IvanZosimov) [reply __tests__/installer.test.ts:250]: Done.

**Review by MaksimZhukov — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**